### ISSUE TITLE
[V2] Incompatible with VRSettings.renderViewportScale changes at runtime

### ISSUE THREAD

**Issue by go1dfish** [Unity-Technologies/PostProcessing#102]
Most all screen effects in unity right now break in VR when the renderViewportScale changes at runtime as happens with the [VRTK Adaptive Renderer](https://vrtoolkit.readme.io/docs/vrtk_adaptivequality) or the [Valve Lab renderer](https://www.assetstore.unity3d.com/en/#!/content/63141) on which it is based.

It would be nice if the Post Processing stack was an exception (Currently it breaks with the viewport not matching the headset causing the circles to be visibly misaligned)

Unfortunately shader dev is still beyond my skills, but the link above has some information on getting effects to work:

> Some shaders, especially Image Effects, need to be modified to work with the changed render scale. To fix them
pass 1.0f / VRSettings.renderViewportScale into the shader and scale all incoming UV values with it in the vertex
program. Do this by using Material.SetFloat to set the value in the script that configures the shader.
>
> In the .shader file: Add a new runtime-set property value float _InverseOfRenderViewportScale and add vertexInput.texcoord *= _InverseOfRenderViewportScale to the start of the vertex program
> In the .cs file: Before using the material (eg. Graphics.Blit) add material.SetFloat("_InverseOfRenderViewportScale", 1.0f / VRSettings.renderViewportScale)

**Comment by JashanChittesh** [comment]
Any news on this one?

At the moment, either PostProcessing or VRSettings.renderViewportScale is severely broken. I've tried applying the fix suggested by the VRTK devs but this didn't really help:

In one case, while the postprocessing stack was active, I could change VRSettings.renderViewportScale just fine, with the intended effect (and without the suggested change applied). In that case, having that change applied would make the image super-small on the HMD.

In another case (just changed the level - probably this did activate fog in the postprocessing stack), the "patch" would be neccesary, without it, the odd scaling occurred. But, a) the patch only worked for the geometry and broke the UI (the UI was fine without the patch), and b) color grading is severely broken with all colors being completed messed up.

This is on Unity 5.6.2p1, using a fairly current version of the PostProcessing stack (v1). Effects I'm using are Fog (only sometimes), Bloom, Colorgrading.

**Comment by bddckr** [comment]
If you mean the issue is Unity only rendering a part of the image and stretching that one to fill the whole view: It looks like that issue was added again in 5.6.1p4 after being a problem since the 5.6 beta days. It has been fixed several times since those betas, which makes me wonder why it's in again.

I'm not using the PP Stack and still got these issues, so I think it's 5.6.1p4 and probably 5.6.2p1, too (didn't try the latter yet).

**Comment by bddckr** [comment]
It looks like 2017.2.0b10 doesn't have the `renderViewportScale` issue by default anymore, it's only there with the PP stack turned on.

Any news on this?

**Comment by SmartCarrion** [comment]
is there a Unity bug we can all vote on to help resolve this renderViewportScale issue?  All the ones I see appear to be closed :-(

**Comment by bddckr** [comment]
That closed one was the one we used previously, but as it's a regression someone has to report it again.
I don't have the time needed to make the Unity engineers actually be fine with a report, but maybe they accept a report without any test project etc. given that it's a regression.

So if you wanna open a new issue with them go ahead, make sure to link the old issues you found so they know that it's actually a regression. And please link said issue on here so we can upvote.

**Comment by robbiesri** [comment]
Hey all!

Admittedly, I'm not up-to-date on all the RVS + PP issues out there, but I did just see this.  I'm working on supporting RVS in Post Processing Stack v2, and I'm working on landing the change soon.

**Comment by robbiesri** [comment]
Whoa, I totally forgot to check in here.  RVS should be supported in Post-processing Stack v2 since I landed the PR, in November 2017.

**Comment by jimmikaelkael** [comment]
@robbiesri Just tested the lastest PPS v2 with Unity 2017.3.1f1 and there still some problems when renderViewportScale is not 1. I'd like to help to solve this as I have the exact same problem for the HBAO asset I distribute on the UAS.

**Comment by robbiesri** [comment]
What's the setup that you are using RVS with? 

My biggest impediment right now is that I don't have a good test setup for the moment, so I can't actually do active dev on this stuff -_-.  I'm hoping to get unblocked sooner than later.

**Comment by jimmikaelkael** [comment]
@robbiesri I'm testing with Oculus CV1 HMD throught SteamVR, Win7 x64. One of my customer have the same problem on his Vive (both with PPS v2 and my asset).
I can do all the necessary test, this is not a problem for me.

**Comment by robbiesri** [comment]
Sorry I wasn't clear.  I'm looking for what XR render mode (multi or single pass) and what effects.

**Comment by jimmikaelkael** [comment]
@robbiesri MultiPass or SinglePass exhibit the exact same problem and I was testing Ambient Occlusion. 
There's some sort of flat AO matrix floating around the scene (hard to describe what you see in the HMD :)).

**Comment by robbiesri** [comment]
Ah, I see. Ok, some history.  When I was working on this, only SAO existed in the repo.  MSVO was under active dev, so I refrained from intersecting with it.  MSVO is the default now, and it wasn't tested against RVS, so it's reasonable that it has some artifacts.  MSVO is on the long list of stuff to inspect for single-pass and other modes.

**Comment by jimmikaelkael** [comment]
@robbiesri hmmm, I think I defined ScalableAO in the ambient occlusion settings. I'll double check this on report after lunch.

**Comment by jimmikaelkael** [comment]
@robbiesri So, I checked it. ScalableAO exhibit artifacts when renderViewportScale isn't 1. I deliberatly exagerated AO intensity.

![capture](https://user-images.githubusercontent.com/414088/38988343-1f6df856-43d3-11e8-9331-397648602a44.JPG)

**Comment by robbiesri** [comment]
Nooooo, why.  Ok, thanks so much for this helpful image.  Now...once I get a viable setup, I can work on this again 😢

**Comment by jimmikaelkael** [comment]
@robbiesri If you have any idea of what's going wrong don't hesitate to send me a diff file as I can test for you and report back.
On my side I tried to look for breaking changes but didn't found anything relevant...

Concerning the bug: lower is the renderViewportScale, higher is the AO offset to upper right corner.

**Comment by angularsen** [comment]
@robbiesri @jimmikaelkael Did you ever resolve this? 

I am having severe issues when using Oculus SDK and `OVRManager`'s adaptive resolution with Max Render Scale > 1.

As the adaptive resolution changes at runtime, there are scaling issues in both the game view and in the Oculus Rift headset:

![image](https://user-images.githubusercontent.com/787816/71384132-e352a380-25df-11ea-9d44-4fc18c406662.png)

I can only reproduce with Oculus SDK.
I can not reproduce with OpenVR SDK.

Some more details here:
https://forum.unity.com/threads/single-pass-stereo-rendering-causes-double-vision.529815/#post-5309499

I'm quite surprised to find and old thread on this and no updates. I see it is marked as closed, but hoping you could take another look at it?
